### PR TITLE
chore: add missing transformer sent stat

### PR DESCRIPTION
**Pull request by cisse21**
# Description

Add missing `sentStat` in `transformer.go`

Right now if the response from transformer has mismatch in the contract we expect we end up discarding those jobs to proc errors but ideally there should be no breach in contract and if there is a breach in contract we should stop processing and fix the same. Since the current code structure does not allow propagating errors, adding a panic to stop processing

## Linear Ticket

Fixes PIPE-1248

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

### CODE REVIEW

File: processor/transformer/transformer.go [hunk 1]
```diff
@@ -395,27 +393,22 @@ func (trans *handle) request(ctx context.Context, url, stage string, data []Tran
...
+		// Panic the processor to avoid replays
 		if err != nil {
 			trans.logger.Errorf("Data sent to transformer : %v", string(rawJSON))
 			trans.logger.Errorf("Transformer returned : %v", string(respData))
-
-			respData = []byte(fmt.Sprintf("Failed to unmarshal transformer response: %s", string(respData)))
-
-			transformerResponses = nil
-			statusCode = http.StatusBadRequest
+			panic(err)
```
Comment (fracasula) [inline processor/transformer/transformer.go:409]: PR description and title are quite misleading, right? You're not just adding a missing stat.
How come we are panicking now? Can you add some context in the PR description please?

Comment (cisse21) [reply processor/transformer/transformer.go:409]: I just added some clean up too .. Will add description for the same

File: processor/transformer/transformer.go [hunk 2]
```diff
@@ -270,6 +270,7 @@ func (trans *handle) transform(
 		trackWg.Done()
 	}()
 
+	trans.sentStat.Count(len(clientEvents))
```
Comment (Sidddddarth) [inline processor/transformer/transformer.go:273]: we could do this after actually sending the events

**Review by Sidddddarth — APPROVED** [review]

**Review by ktgowtham — APPROVED** [review]
LGTM.

**Pull request merged**